x-pack/test/performance/services/playwright.ts

```diff
@@ -0,0 +1,256 @@
+    return (this.currentSpanStack.length ? this.currentSpanStack[0] : this.currentTransaction)
+      ?.traceparent;
+  }
+
+  private async getStorageState() {
+    if (this.storageState) {
+      return this.storageState;
+    }
+
+    await this.withSpan('initial login', undefined, async () => {
+      const kibanaUrl = Url.format({
+        protocol: this.config.get('servers.kibana.protocol'),
+        hostname: this.config.get('servers.kibana.hostname'),
+        port: this.config.get('servers.kibana.port'),
+      });
+
+      const browser = await this.getBrowserInstance();
+      const context = await browser.newContext();
+      const page = await context.newPage();
+      await this.interceptBrowserRequests(page);
+      await page.goto(`${kibanaUrl}`);
+
+      const usernameLocator = page.locator('[data-test-subj=loginUsername]');
+      const passwordLocator = page.locator('[data-test-subj=loginPassword]');
+      const submitButtonLocator = page.locator('[data-test-subj=loginSubmit]');
+
+      const noDelayOnUserActions = process.env.TEST_DONT_DELAY_USER_ACTIONS === 'true';
+
+      await usernameLocator?.type('elastic', { delay: noDelayOnUserActions ? 0 : 500 });
+      await passwordLocator?.type('changeme', { delay: noDelayOnUserActions ? 0 : 500 });
+      await submitButtonLocator?.click({ delay: noDelayOnUserActions ? 0 : 1000 });
+
+      await page.waitForSelector('#headerUserMenu');
+
+      this.storageState = await page.context().storageState();
+      await page.close();
+      await context.close();
+    });
+
+    return this.storageState;
+  }
+
+  private async getBrowserInstance() {
+    if (this.browser) {
+      return this.browser;
+    }
+    return await this.withSpan('browser creation', 'setup', async () => {
+      const headless = !!(process.env.TEST_BROWSER_HEADLESS || process.env.CI);
+      this.browser = await playwright.chromium.launch({ headless, timeout: 60000 });
+      return this.browser;
+    });
+  }
+
+  private async sendCDPCommands(context: BrowserContext, page: Page) {
+    const client = await context.newCDPSession(page);
+
+    await client.send('Network.clearBrowserCache');
+    await client.send('Network.setCacheDisabled', { cacheDisabled: true });
+    await client.send('Network.emulateNetworkConditions', {
+      latency: 100,
+      downloadThroughput: 750_000,
+      uploadThroughput: 750_000,
+      offline: false,
+    });
+
+    return client;
+  }
+
+  private async interceptBrowserRequests(page: Page) {
+    await page.route('**', async (route, request) => {
+      const headers = await request.allHeaders();
+      const traceparent = this.getCurrentTraceparent();
+      if (traceparent && request.isNavigationRequest()) {
+        await route.continue({ headers: { traceparent, ...headers } });
+      } else {
+        await route.continue();
+      }
+    });
+  }
+
+  public makePage({ journeyName, autoLogin }: { journeyName: string; autoLogin?: boolean }) {
```

Comment:
Nit: Can we rename this to something like

```ts
const { step } = performanceTests.journey(name, { autoLogin?: boolean })
```

Maybe move it to something like we talked about with steps being an option rather than defined after the journey is created?